### PR TITLE
fix: extends clusterRole to allow cronjob and Argo workflows triggers

### LINKED ISSUE #15298: v2.8.2 raw manifests install method is missing jobs RBACs
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**
Whenever installing argoCD via RAW manifests [manifests](https://argo-cd.readthedocs.io/en/stable/#getting-started), the `argocd-server` `ClusterRole` is missing a RBAC to allow the creation of cronjobs via UI. 

**To Reproduce**
1. Install ArgoCD as described at https://argo-cd.readthedocs.io/en/stable/#getting-started
2. Deploy an application which has a `cronjob` resource
3. Try to trigger it via UI

**Expected behavior**

Should work as expected. 

**Screenshots**

image

**Version**

```shell
v2.8
```

**Logs**

```
Paste any relevant application logs here.
argocd-server-c57d6f87d-j2x78 argocd-server time="2023-08-31T02:05:26Z" level=warning msg="finished unary call with code PermissionDenied" error="rpc error: code = PermissionDenied desc = jobs.batch is forbidden: User \"system:serviceaccount:argocd:argocd-server\" cannot create resource \"jobs\" in API group \"batch\" in the namespace \"default\"" grpc.code=PermissionDenied grpc.method=RunResourceAction grpc.service=application.ApplicationService grpc.start_time="2023-08-31T02:05:26Z" grpc.time_ms=74.363 span.kind=server system=grpc
```

### PR DESCRIPTION
**Pull request by tchellomello**
Fixes: https://github.com/argoproj/argo-cd/issues/15298

Extends the argocd-server `ClusterRole` to allow triggering `cronjobs` and Argo `workflows` via UI. 

Similar to: https://github.com/argoproj/argo-helm/pull/2212/files

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

### CODE REVIEW

File: manifests/cluster-rbac/server/argocd-server-clusterrole.yaml [hunk 1]
```diff
@@ -36,3 +36,15 @@ rules:
...
+  resources:
+  - jobs
+  verbs:
+  - create   # supports triggering jobs from UI
+- apiGroups:
+  - argoproj.io
+  resources:
+  - workflows
+  verbs:
+  - create   # supports triggering jobs from UI
```
Comment (morey-tech) [inline manifests/cluster-rbac/server/argocd-server-clusterrole.yaml:50]:
```suggestion
  - create   # supports triggering workflows from UI
```

**Review by morey-tech — APPROVED** [review]
Good catch! Looks like that was missed in https://github.com/argoproj/argo-cd/pull/12925. The helm chart implemented the same fix on their end https://github.com/argoproj/argo-helm/pull/2212

**Review by 34fathombelow — APPROVED** [review]
LGTM, awesome find!

File: manifests/cluster-rbac/server/argocd-server-clusterrole.yaml [hunk 2]
```diff
@@ -36,3 +36,15 @@ rules:
...
+  resources:
+  - jobs
+  verbs:
+  - create   # supports triggering jobs from UI
+- apiGroups:
+  - argoproj.io
+  resources:
+  - workflows
+  verbs:
+  - create   # supports triggering workflows from UI
```
Comment (terrytangyuan) [inline manifests/cluster-rbac/server/argocd-server-clusterrole.yaml:50]: Should this be in a separate PR? Or maybe update PR title and description to include this change

Comment (tchellomello) [reply manifests/cluster-rbac/server/argocd-server-clusterrole.yaml:50]: Sure it makes sense. I will update the description and title shortly.

Comment (tchellomello) [reply manifests/cluster-rbac/server/argocd-server-clusterrole.yaml:50]: @terrytangyuan commit and description updated. Thanks!

**Review by terrytangyuan — APPROVED** [review]

**Review by morey-tech — APPROVED** [review]

**Comment by tchellomello** [comment]
@terrytangyuan @34fathombelow @morey-tech Thanks so much for your time reviewing the PR. Guys, is there anything else is needed on my end to get this merged? Thank you so much!

**Comment by terrytangyuan** [comment]
We'll need someone with approval status to perform final approve and merge: https://github.com/argoproj/argo-cd/blob/master/OWNERS#L5-L16

**Review by jannfis — APPROVED** [review]
LGTM
